File: frontend/lib/model/team.dart
/// Author: Łukasz Piętka (FUT 2025)
class Team {
  final int id;
  final String name;
  final String captainName;
  final List<String>? members;
  final String imageUrl;
  final int points;
  final DateTime? lastModified;

  Team({
    this.id = -1,
    required this.name,
    this.captainName = "Brak",
    this.members,
    required this.imageUrl,
    this.points = 0,
    this.lastModified
  });
}

File: backend/src/main/java/me/havooc/poligon/model/Team.java
/**
 * @Author Łukasz Piętka (FUT 2025)
 */
package me.havooc.poligon.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Team {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String captainName;
    @ElementCollection(fetch = FetchType.EAGER)
    private List<String> members;
    private String imageUrl;
    private int points;
    @UpdateTimestamp
    private LocalDateTime lastModified;
}